mlp.py

```diff
@@ -0,0 +1,130 @@
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import OneHotEncoder
+import numpy as np
+from utils import *
+
+
+class MultiLayerPerceptron:
+    def __init__(self, layers):
+        self.layers = layers
+
+        self.params = {}
+
+    def initialize_parameters(self):
+        for l in range(1, len(self.layers)):
+            in_size = self.layers[l - 1]
+            out_size = self.layers[l]
+
+            self.params["w" + str(l)] = np.random.randn(out_size, in_size) / np.sqrt(
+                in_size
+            )
+            self.params["b" + str(l)] = np.zeros((out_size, 1))
+
+    def model_forward(self, X):
+        number_of_layers = len(self.layers) - 1
+        a = X
+        caches = {}
+
+        for l in range(1, number_of_layers):
+            a_prev = a
+            weight_string = "w" + str(l)
+            bias_string = "b" + str(l)
```

Comment:
**Suggestion:** Use f-string formatting for generating weight and bias strings. This will make the code more readable and concise.
```suggestion
            weight_string = f"w{l}"
            bias_string = f"b{l}"
```